### PR TITLE
reduce signal_pipe_buf_size 

### PR DESCRIPTION
**Pull request by save-my-heart**
Changelog category (leave one):
- Not for changelog

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

- ensure the `signal_pipe_buf_size` is <= `PIPE_BUF`.
- extract query_id & query from `ThreadStatus` if possible.

### CODE REVIEW

**Comment by save-my-heart** [comment]
@azat

File: base/daemon/BaseDaemon.cpp [hunk 1]
```diff
@@ -279,18 +271,27 @@ class SignalListener : public Poco::Runnable
...
         DB::ThreadStatus thread_status;
 
+        String query_id;
+        String query;
+
         /// Send logs from this thread to client if possible.
         /// It will allow client to see failure messages directly.
         if (thread_ptr)
         {
+            query_id = thread_ptr->getQueryId().toString();
```
Comment (azat) [inline base/daemon/BaseDaemon.cpp:288]: Each extra memory read may lead to SIGSEGV from the SIGSEGV handler, but I guess this is fine to obtain query_id/query from the thread_ptr.

File: base/daemon/BaseDaemon.cpp [hunk 2]
```diff
@@ -389,20 +390,16 @@ static void sanitizerDeathCallback()
...
     DB::writePODBinary(stack_trace, out);
     DB::writeBinary(UInt32(getThreadId()), out);
-    DB::writeStringBinary(query_id, out);
     DB::writePODBinary(DB::current_thread, out);
 
     out.next();
 
     /// The time that is usually enough for separate thread to print info into log.
-    sleepForSeconds(10);
+    sleepForSeconds(20);
```
Comment (azat) [inline base/daemon/BaseDaemon.cpp:402]: What is the reason for doubling this?

**Review by azat — APPROVED** [review]
LGTM, now you should wait for someone to set `can-be-tested` label.

File: base/daemon/BaseDaemon.cpp (hunk 2 above)
Comment (save-my-heart) [reply base/daemon/BaseDaemon.cpp:402]: > What is the reason for doubling this?

Same to `signalHandler`, which may be not necessary  at all.

https://github.com/ClickHouse/ClickHouse/blob/9e91a9dfd1dae8072d9d2132a4b3e4bbb70e1c1d/base/daemon/BaseDaemon.cpp#L150-L152

Comment (azat) [reply base/daemon/BaseDaemon.cpp:402]: Ok, I see, it had been increased in #19317

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**